src/main/java/com/microsoft/jenkins/kubernetes/KubernetesClientWrapper.java

```diff
@@ -134,55 +116,38 @@ public void apply(FilePath[] configFiles) throws IOException, InterruptedExcepti
+            ResourceManager extensionV1beta1ResourceManager = new ExtensionV1beta1ResourceManager();
+
             // Process the Namespace in the list first, as it may be a dependency of other resources.
-            Iterator<HasMetadata> iter = resources.iterator();
-            while (iter.hasNext()) {
-                HasMetadata resource = iter.next();
-                if (resource instanceof Namespace) {
-                    Namespace namespace = (Namespace) resource;
-                    new NamespaceUpdater(namespace).createOrApply();
-                    iter.remove();
+            Iterator<Object> iterator = resources.iterator();
+            while (iterator.hasNext()) {
+                Object resource = iterator.next();
+                if (resource instanceof V1Namespace) {
+                    v1ResourceManager.apply(resource);
+                    iterator.remove();
                 }
             }
 
-            for (HasMetadata resource : resources) {
-                if (resource instanceof Deployment) {
-                    Deployment deployment = (Deployment) resource;
-                    new DeploymentUpdater(deployment).createOrApply();
-                } else if (resource instanceof Service) {
-                    Service service = (Service) resource;
-                    new ServiceUpdater(service).createOrApply();
-                } else if (resource instanceof Ingress) {
-                    Ingress ingress = (Ingress) resource;
-                    new IngressUpdater(ingress).createOrApply();
-                } else if (resource instanceof ReplicationController) {
-                    ReplicationController replicationController = (ReplicationController) resource;
-                    new ReplicationControllerUpdater(replicationController).createOrApply();
-                } else if (resource instanceof ReplicaSet) {
-                    ReplicaSet rs = (ReplicaSet) resource;
-                    new ReplicaSetUpdater(rs).createOrApply();
-                } else if (resource instanceof DaemonSet) {
-                    DaemonSet daemonSet = (DaemonSet) resource;
-                    new DaemonSetUpdater(daemonSet).createOrApply();
-                } else if (resource instanceof Job) {
-                    Job job = (Job) resource;
-                    new JobUpdater(job).createOrApply();
-                } else if (resource instanceof Pod) {
-                    Pod pod = (Pod) resource;
-                    new PodUpdater(pod).createOrApply();
-                } else if (resource instanceof Secret) {
-                    Secret secret = (Secret) resource;
-                    new SecretUpdater(secret).createOrApply();
-                } else if (resource instanceof ConfigMap) {
-                    ConfigMap configMap = (ConfigMap) resource;
-                    new ConfigMapUpdater(configMap).createOrApply();
-                } else {
+            for (Object resource : resources) {
```

Comment:
Instead of maintaining a long long if/else logic in the `KubernetesClientWrapper.java` and 3 ResourceManager instances, I would like to maintain a collection or a switch of <io.kubernetes.client.models.*.class, CorrespodingResourceUpdater.class>, then find the resource's corresponding resourceUpdater and then call the `createOrApply()`